app/controllers/public_profiles_controller.rb

```diff
@@ -0,0 +1,14 @@
+class PublicProfilesController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    @developer = developer
+  end
+
+  private
+
+  def developer
+    finder = Developers::Finder.new(id: params[:developer_id])
+    finder.developer
```

Comment:
We can combine these lines.

```suggestion
    Developers::Finder.new(id: params[:developer_id]).developer
```